components/common/historyEvents.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// This file adds three CustomEvents to the window object:
+// 1. historychange (when replaceState or pushState is called)
+// 2. replacestate (when replaceState is called)
+// 3. pushstate (when pushState is called)
```

Comment:
It's weird that we don't have these events. Is there any proposal for it?

Reply:
No.... there almost are but I guess it kinda makes sense - the developer is the one in control of when this sort of stuff happens so maybe there's no need for an event?

Personally I think there should be, but I'm not involved with standards anymore :laughing:

Reply:
Actually, I'm wrong: https://github.com/WICG/proposals/issues/6#issuecomment-771827306

Looks like people want it, but its stalled :laughing:

Reply:
It's shipped :fearful:

Reply:
Updated, mind having another look! Thanks for bringing that up, I had no idea it'd been fixed!